great_tables/gt.py

```diff
@@ -314,6 +320,98 @@ def _render_formats(self, context: str) -> Self:
         new_body.render_formats(self._tbl_data, self._substitutions, context)
         return self._replace(_body=new_body)
 
+    def _perform_col_merge(self) -> Self:
+        # If no column merging defined, return unchanged
+        if not self._col_merge:
+            return self  # pragma: no cover
+
+        # Create a copy of the body for modification
+        new_body = self._body.copy()
+
+        # Process each column merge operation in order
+        for col_merge in self._col_merge:
+            if col_merge.type != "merge":
+                # TODO: incorporate other specialized merging operations (e.g., "merge_range") but
+                # only handle the basic 'merge' type for now
+                continue
+
+            # Get the target column (column that receives the merged values)
+            target_column = col_merge.vars[0]
+
+            # Get all columns, rows, and the pattern for this merge operation
+            columns = col_merge.vars
+            rows = col_merge.rows
+            pattern = col_merge.pattern
+
+            # Pattern should always be set by `.cols_merge()`, but check here just in case
+            if pattern is None:  # pragma: no cover
+                raise ValueError("Pattern must be provided for column merge operations.")
+
+            # Validate that pattern references are valid
+            pattern_cols = _extract_pattern_columns(pattern)
+
+            # With each column reference in the pattern, check that it is valid
+            for col_ref in pattern_cols:
+                # The pattern syntax uses 1-based indexing so adjust here
```

Comment:
1-based indexing seems like a bad idea (but maybe it happens in other places in Python?)